source/daq/frequency_mask_trigger.cc

```diff
@@ -51,27 +54,48 @@ namespace psyllid
         f_n_packets_for_mask = a_n_pkts;
         return;
     }
-
     void frequency_mask_trigger::set_threshold_ampl_snr( double a_ampl_snr )
     {
         f_threshold_snr = a_ampl_snr * a_ampl_snr;
         LDEBUG( plog, "Setting threshold (power via ampl) to " << f_threshold_snr );
         return;
     }
-
     void frequency_mask_trigger::set_threshold_power_snr( double a_power_snr )
     {
         f_threshold_snr = a_power_snr;
         LDEBUG( plog, "Setting threshold (power via power) to " << f_threshold_snr );
         return;
     }
-
+    void frequency_mask_trigger::set_threshold_power_snr_high( double a_power_snr )
+    {
+        f_threshold_snr_high = a_power_snr;
+        LDEBUG( plog, "Setting threshold (power via power) to " << f_threshold_snr_high );
+        return;
+    }
     void frequency_mask_trigger::set_threshold_dB( double a_dB )
     {
         f_threshold_snr = pow( 10, a_dB / 10. );
         LDEBUG( plog, "Setting threshold (power via dB) to " << f_threshold_snr );
         return;
     }
+    void frequency_mask_trigger::set_trigger_mode( unsigned trigger_mode_id )
```

Comment:
I didn't know how to send a trigger_mode_t type variable to/from dragonfly.
Therefore the set method gets an unsigned int (1 or 2) and sets both the trigger_mode_id and the trigger_mode.
When the trigger-mode is requested psyllid returns the trigger_mode_id.